hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeStateMachine.java

```diff
@@ -195,6 +199,14 @@ private void start() throws IOException {
         LOG.error("Unable to finish the execution.", e);
       }
     }
+
+    // If we have got some exception in stateMachine we set the state to
+    // shutdown to stop the stateMachine thread. Along with this we should
+    // also stop the datanode.
+    if (context.getShutdownStateMachine()) {
```

Comment:
What happens if we transition to shutdown state after this check? Is process shutdown done then?

Reply:
If this check is true means, the state is transitioned to shutdown, we call stop() and terminate. So, datanode process is terminated.